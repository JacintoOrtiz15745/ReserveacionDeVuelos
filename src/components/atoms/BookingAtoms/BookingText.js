import React from 'react';
import {Text, View} from 'react-native';
import {styles} from '../../../styles/BookingStyles';
const BookingText = ({title, subtitle}) => {
  return (
    <View testID="view">
      <Text style={styles.InformationFlightsTitle} testID="title">
        {title}
      </Text>
      <Text style={styles.InformationFlightsSubtitle} testID="subtitle">
        {subtitle}
      </Text>
    </View>
  );
};

export default BookingText;
