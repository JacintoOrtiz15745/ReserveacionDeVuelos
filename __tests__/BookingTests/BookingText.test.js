import React from 'react';
import {shallow} from 'enzyme';
import BookingText from '../../src/components/atoms/BookingAtoms/BookingText';

describe('component renders correctly', () => {
  const props = {
    title: 'title',
    subtitle: 'subtitle',
  };
  const wrapper = shallow(<BookingText {...props} />);
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('component exists', () => {
    const component = wrapper.findWhere(
      (node) => node.prop('testID') === 'view',
    );
    expect(component.exists()).toBeTruthy();
  });
  test('component exists', () => {
    const title = wrapper.findWhere((node) => node.prop('testID') === 'title');
    expect(title.props().children).toEqual(props.title);
  });
  test('component exists', () => {
    const subtitle = wrapper.findWhere(
      (node) => node.prop('testID') === 'subtitle',
    );
    expect(subtitle.props().children).toEqual(props.subtitle);
  });
});
