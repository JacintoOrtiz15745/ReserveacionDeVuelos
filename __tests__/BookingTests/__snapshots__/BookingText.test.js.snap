// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly snapshot 1`] = `
<View
  testID="view"
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
    testID="title"
  >
    title
  </Text>
  <Text
    style={
      Object {
        "color": "#b6b7ba",
        "marginTop": 5,
      }
    }
    testID="subtitle"
  >
    subtitle
  </Text>
</View>
`;
